src/main/java/de/tum/in/www1/artemis/service/exam/ExamRegistrationService.java

```diff
@@ -74,30 +81,53 @@ public ExamRegistrationService(ExamRepository examRepository, UserService userSe
      *
-     * @param courseId    the id of the course
-     * @param examId      the id of the exam
-     * @param studentDTOs the list of students (with at least registration number) who should get access to the exam
+     * @param courseId     the id of the course
+     * @param examId       the id of the exam
+     * @param examUserDTOs the list of students (with at least registration number) who should get access to the exam
      * @return the list of students who could not be registered for the exam, because they could NOT be found in the Artemis database and could NOT be found in the TUM LDAP
      */
-    public List<StudentDTO> registerStudentsForExam(Long courseId, Long examId, List<StudentDTO> studentDTOs) {
+    public List<ExamUserDTO> registerStudentsForExam(Long courseId, Long examId, List<ExamUserDTO> examUserDTOs) {
         var course = courseRepository.findByIdElseThrow(courseId);
-        var exam = examRepository.findWithRegisteredUsersById(examId).orElseThrow(() -> new EntityNotFoundException("Exam", examId));
+        var exam = examRepository.findWithExamUsersById(examId).orElseThrow(() -> new EntityNotFoundException("Exam", examId));
 
         if (exam.isTestExam()) {
             throw new AccessForbiddenException("Registration of students is only allowed for real exams");
         }
 
-        List<StudentDTO> notFoundStudentsDTOs = new ArrayList<>();
-        for (var studentDto : studentDTOs) {
-            var registrationNumber = studentDto.getRegistrationNumber();
-            var login = studentDto.getLogin();
-            var email = studentDto.getEmail();
+        List<ExamUserDTO> notFoundStudentsDTOs = new ArrayList<>();
+        for (var examUserDto : examUserDTOs) {
+            var registrationNumber = examUserDto.registrationNumber();
+            var login = examUserDto.login();
+            var email = examUserDto.email();
             Optional<User> optionalStudent = userService.findUserAndAddToCourse(registrationNumber, course.getStudentGroupName(), Role.STUDENT, login, email);
             if (optionalStudent.isEmpty()) {
-                notFoundStudentsDTOs.add(studentDto);
+                notFoundStudentsDTOs.add(examUserDto);
             }
             else {
-                exam.addRegisteredUser(optionalStudent.get());
+                User student = optionalStudent.get();
+                ExamUser examUser = examUserRepository.findByExamIdAndUserId(exam.getId(), student.getId());
+                if (!exam.getExamUsers().contains(examUser) && !authorizationCheckService.isInstructorInCourse(course, student) && !authorizationCheckService.isAdmin(student)) {
+                    ExamUser registeredExamUser = new ExamUser();
+                    registeredExamUser.setUser(optionalStudent.get());
+                    registeredExamUser.setExam(exam);
+
+                    if (!StringUtils.isBlank(examUserDto.room())) {
```

Comment:
Please don't import StringUtils from this weird package and instead check for null and use `string.isBlank()` from Java 11.